sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/CommunicationIdentityClientTestBase.java

```diff
@@ -19,41 +15,54 @@
 
 public class CommunicationIdentityClientTestBase extends TestBase {
     protected static final TestMode TEST_MODE = initializeTestMode();
-
     protected static final String ENDPOINT = Configuration.getGlobalConfiguration()
-        .get("ADMINISTRATION_SERVICE_ENDPOINT", "https://yourresource.communication.azure.com");
+        .get("ADMINISTRATION_SERVICE_ENDPOINT", "https://REDACTED.communication.azure.com");
+
+    protected static final String ACCESSKEYRAW = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
+    protected static final String ACCESSKEYENCODED = Base64.getEncoder().encodeToString(ACCESSKEYRAW.getBytes());
+    protected static final String ACCESSKEY = Configuration.getGlobalConfiguration()
+        .get("ADMINISTRATION_SERVICE_ACCESS_TOKEN", ACCESSKEYENCODED);
 
-    protected static final String MOCKACCESSTOKEN = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
-    protected static final String MOCKACCESSTOKENENCODED = Base64.getEncoder().encodeToString(MOCKACCESSTOKEN.getBytes());
-    protected static final String ACCESSTOKEN = Configuration.getGlobalConfiguration()
-        .get("ADMINISTRATION_SERVICE_ACCESS_TOKEN", MOCKACCESSTOKENENCODED);
+    protected static final String CONNECTION_STRING = Configuration.getGlobalConfiguration()
+        .get("COMMUNICATION_CONNECTION_STRING", "endpoint=https://REDACTED.communication.azure.com/;accesskey=" + ACCESSKEYENCODED);
     
     protected CommunicationIdentityClientBuilder getCommunicationIdentityClient() {
-        try {
-            CommunicationClientCredential credential = new CommunicationClientCredential(ACCESSTOKEN);
-            CommunicationIdentityClientBuilder builder = new CommunicationIdentityClientBuilder();
-            builder.endpoint(ENDPOINT)
-                .credential(credential);
-    
-            if (interceptorManager.isPlaybackMode()) {
-                builder.httpClient(interceptorManager.getPlaybackClient());
-                return builder;
-            } else {
-                HttpClient client = new NettyAsyncHttpClientBuilder().build();
-                builder.httpClient(client);
-            }
-    
-            if (!interceptorManager.isLiveMode()) {
-                builder.addPolicy(interceptorManager.getRecordPolicy());
-            }
-    
+        CommunicationIdentityClientBuilder builder = new CommunicationIdentityClientBuilder();
+        builder.endpoint(ENDPOINT)
+            .accessKey(ACCESSKEY);
+
+        if (interceptorManager.isPlaybackMode()) {
+            builder.httpClient(interceptorManager.getPlaybackClient());
+            return builder;
+        } else {
+            HttpClient client = new NettyAsyncHttpClientBuilder().build();
```

Comment:
TestBase class has a method called `getHttpClients()` which returns a list of all available HTTP clients. You can use that method instead. That will allow you to test your service with multiple HTTP clients.

Reply:
How would this work since `builder.httpClient(client);` only takes in one client?